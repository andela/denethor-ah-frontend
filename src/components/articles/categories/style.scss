.feed-bottom-section {
  margin-top: 6rem;
}

.search-error {
  height: 80px;
  width: 100%;
  justify-content: center;
  margin-top: 20%;
  margin-bottom: 15%;
  min-height: 20em;
 }