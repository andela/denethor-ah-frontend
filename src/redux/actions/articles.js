import {
  ADD_ARTICLE,
  REMOVE_ARTICLE,
  GET_ONE_ARTICLE_SUCCESS,
  GET_ARTICLES_SUCCESS,
  RATE_ARTICLE_SUCCESS,
  GET_ARTICLE_AVERAGE_RATING_SUCCESS,
  LIKE_ARTICLE_SUCCESS,
  DISLIKE_ARTICLE_SUCCESS,
  BOOKMARK_ARTICLE_SUCCESS,
} from './types';
import axios from '../../utils/axiosConfig';
import { extractImageFromBody } from '../../utils/imageExtractor';
import { toTimeFromNow } from '../../utils/dateTime';

const { API_ROOT_URL } = process.env;

export const addArticle = article => (dispatch) => {
  dispatch({
    type: ADD_ARTICLE,
    article
  });
};

export const removeArticle = id => (dispatch) => {
  dispatch({
    type: REMOVE_ARTICLE,
    id
  });
};

export const getArticles = (category) => (dispatch) => {
  const query = category && category !== undefined ? `category=${category}` : ''
  return axios
  .get(`${API_ROOT_URL}/articles/?${query}`)
  .then((response) => {
    const articles = response.data.data.map(article => {
      article.featuredImage = extractImageFromBody(article.body);
      article.dateCreated =  toTimeFromNow(article.createdAt);
      return article;
    });
    return dispatch({
      type: GET_ARTICLES_SUCCESS,
      payload: articles
    });
  })
  .catch(error => {
    throw error;
  })
}

export const getOneArticle = id => (dispatch) => {
  return axios.get(`${API_ROOT_URL}/articles/${id}`)
  .then(response => {
    const article = { ...response.data.data };
    article.featuredImage = extractImageFromBody(article.body);
    article.dateCreated =  toTimeFromNow(article.createdAt);
    dispatch({
      type: GET_ONE_ARTICLE_SUCCESS,
      payload: article
    });
  })
  .catch(error => {
    throw error;
  });
};

export const rateArticle = (rating, articleId) => (dispatch) => {
  return axios
  .post(`${API_ROOT_URL}/articles/${articleId}/ratings`, {rating})
  .then(response => {
    dispatch({
      type: RATE_ARTICLE_SUCCESS,
      payload: response.data.article
    })
  })
  .catch((error) => {
    throw error
  })
}

export const getArticleAvgRating = (articleId) => (dispatch) => {
  return axios
  .get(`${API_ROOT_URL}/articles/${articleId}/ratings`)
  .then(response => {
    const { count, rows }= response.data.data
    dispatch({
      type: GET_ARTICLE_AVERAGE_RATING_SUCCESS,
      payload: {
        count, rows, articleId
      }
    })
  })
  .catch((error) => {
    throw error
  })
}

export const likeArticle = (articleId) => (dispatch) => {
  return axios
  .patch(`${API_ROOT_URL}/articles/${articleId}/likes`, {})
  .then(response => {
    dispatch({
      type: LIKE_ARTICLE_SUCCESS,
      payload: response.data.data.message
    });
    return response
  })
  .catch((error) => {
    throw error
  })
}

export const dislikeArticle = (articleId) => (dispatch) => {
  return axios
  .patch(`${API_ROOT_URL}/articles/${articleId}/dislikes`, {})
  .then(response => {
    dispatch({
      type: DISLIKE_ARTICLE_SUCCESS,
      payload: response.data.data.message
    });
    return response
  })
  .catch((error) => {
    throw error
  })
}

export const bookmarkArticle = (articleId) => (dispatch) => {
  return axios
  .post(`${API_ROOT_URL}/articles/${articleId}/bookmark`)
  .then(response => {
    console.log(response)
    dispatch({
      type: BOOKMARK_ARTICLE_SUCCESS,
      payload: response.data
    })
    return response
  })
  .catch((error) => {
    throw error
  })
}
