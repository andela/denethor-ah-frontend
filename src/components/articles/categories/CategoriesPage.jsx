import React from 'react';
import PropTypes from 'prop-types';
import queryString from 'query-string';
import ServerPaginator from '../../paginator/ServerPaginator';
import './style.scss';

const CategoriesPage = ({ location: { search } }) => {
  const category = queryString.parse(search).category;

  return (
    <div className="feed-bottom-section">
      <ServerPaginator url={`/articles?category=${category}`} />
    </div>  
  )
};

CategoriesPage.propTypes = {
  location: PropTypes.object,
}

export default CategoriesPage;
