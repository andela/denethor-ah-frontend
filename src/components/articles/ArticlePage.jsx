import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import RoundedImage from '../RoundedImage';
import VerticalMargin from '../VerticalMargin';
import HorizontalMargin from '../HorizontalMargin';
import AuthModal from '../authModal';
import { RatingStars, RatingStarsBox } from '../ratingStars';
import { HorizontalLine } from '../horizontalLine';
import { CreateComment, CommentEntries } from '../comments';
import { addComment, getArticleComments, getAllCommentsImpression, likeComment } from '../../redux/actions/comments';
import { 
  getOneArticle, 
  rateArticle, 
  getArticleAvgRating,
  likeArticle, 
  dislikeArticle, 
  bookmarkArticle,
  getArticleDislikes,
  getArticleLikes,
  removeArticleBookmark,
  getUserBookmarks,
} from '../../redux/actions/articles';
import { LikeNumberGroup , DisLikeNumberGroup, CommentButtonGroup } from '../likeNumberGroup';
import { BookMarkHeartIcon } from '../impressionIcons';
import { activateQuickAuthAction } from '../../redux/actions/auth';

import './style.scss';
import Spinner from '../spinner/Spinner';


export class SingleArticleView extends Component {
  state = {
    likeBtnColor: '#822E2E',
    BookMarkBtnColor: '#ffffff',
    currentUserRating: 0,
    showModal: false
  }


  async componentDidMount () {
    const { fetchArticle, getArticleAvgRating, getArticleLikes, getArticleDislikes, getAllCommentsImpression, match,  currentUserId, getUserBookmarks } = this.props;
    let articleId = match.params.articleId;
    
    const { comments } = await fetchArticle(articleId);
    const commentsIds = comments.map(comment => comment.id);
    await getAllCommentsImpression(commentsIds);
    getArticleLikes(articleId);
    getArticleDislikes(articleId);
    getArticleAvgRating(articleId);

    if (currentUserId && !this.state.currentUserId) {
      getUserBookmarks(currentUserId);
      this.setState({ currentUserId });
    }
  }

  activateModal = () => {
    this.setState({ showModal: true });
    this.props.activateQuickAuthAction();
  }

  toggleModalOff = () => {
    this.setState({ showModal: false });
  }

  starClickHandle = (item) => {
    let articleId = this.props.match.params.articleId;
    return this.props.rateArticle(item, articleId )
    .then(() => {
      this.setState({currentUserRating: item})
      toast.success('Thank you for rating this article')}
      ) 
    .catch(error => {
      if (error.response.data.message === 'You already rated this article'){
        return toast.error('You have already rated this article');
      } else if(error.response.data === 'Unauthorized'){
        toast.error('You need to login to rate an article');
        return  this.activateModal();
      }else if (error.response.data.message === "Sorry! You can't rate your article"){
        return toast.error('Sorry! You can\'t rate your article');
      }
      return toast.error('Error while rating article');
    }); 
  }

  likeHandle = () => {
    let articleId = this.props.match.params.articleId;
    return this.props.likeArticle(articleId)
      .then((response) => {
        const message = response.data.data.message;
        toast.success(`${message}`)
      })
      .catch(error => {
        if(error.response && error.response.status === 401){
          toast.error('You need to login to like an article',{
            className: 'toast-custom-style'
          });
          return  this.activateModal();
        }
        toast.error(`An error occurred when trying to like this article`, {
          className: 'toast-custom-style'
        });
      })
  }

  dislikeHandle = () => {
    let articleId = this.props.match.params.articleId;
    return this.props.dislikeArticle(articleId)
      .then((response) => {
        const message = response.data.data.message;
        toast.success(`${message}`)
      })
      .catch(error => {
        if(error.response && error.response.status === 401){
          toast.error('You need to login to dislike an article', {
            className: 'toast-custom-style'
          });
          return  this.activateModal();
      }
        return toast.error('An error occurred while trying to dislike the article', {
          className: 'toast-custom-style'
        });
      })
          
  }

  bookmarkHandle = () => {
    let articleId = this.props.match.params.articleId;
    this.props.bookmarkArticle(articleId).then(() => { this.setState({BookMarkBtnColor: '#ffff00'}) })
   
    .catch(error => {
      if(error.response.status === 401){
        toast.error('You need to login to bookmark an article',{className: 'toast-custom-style'});
        return  this.activateModal(); 
     } else if(error.response.status === 409){
        return this.props.removeArticleBookmark(articleId).then(() => { this.setState({BookMarkBtnColor: '#fff'}) })
     }
      return toast.error('An error occurred while trying to bookmark this article', {
        className: 'toast-custom-style'
      });
    })
  }

  handleOnClickLike = (commentId) => {
    this.props.likeComment(commentId)
    .then()
    .catch(error => {
      const { response } = error;
        if (response && response.status === 422){
          return toast.error(error.response.data.message.details[0].message);
        }
        if(response && response.status === 400){
          return toast.error('You cannot like your own comment');
        }
        if(response && response.status === 401){
          return toast.error('You need to login to like this comment');
        }
        if(response && response.status === 500){
          return toast.error('Server error');
        }
        else {
          return toast.error('unknown error');
        }
    }); 
  }

  render() {
    const {
      userLiked, 
      userDisliked,
      userRating,
      impressions,
      averageRating,
      ratingsCount,
      match: { params: { articleId } },
      articles = [],
      comments,
      addComment,
      quickAuthAction
    } = this.props;

    const { toggleModalOff, state: { showModal } } = this;
    const { loading } = comments;
    const article = articles.find(item => item.id === articleId) || {};
    const commentsCount = this.props.comments.comments.length

    const { 
      title, 
      featuredImage = '/assets/img/map-typing.jpg', 
      author: { username = '', imageUrl: userAvatarImage = '' } = {}, 
      body,
      comments: articleComments = [],
    } = article;

    const totalLikes = impressions.likes.length;
    const totalDislikes = impressions.dislikes.length;

    const { currentAction } = quickAuthAction;
    const authModalContentLookup = {
      'login': 'Login',
      'signup': 'Signup'
    };
    const currentAuthContent =  authModalContentLookup[currentAction];

    return (
      <div>
        {showModal && <AuthModal toggleOff={toggleModalOff} content={currentAuthContent} />}
        <div className='banner-section'>
          <div className='banner-image'>
            <img src={featuredImage} alt='banner background' />
          </div>
          <div className='transparent-overlay' />
          <div className='banner-content'>
            <div className='auto-container'>
              <div className='banner-content-title'>{title}</div>
              <div className='bookmark-button'>
                <BookMarkHeartIcon  onClick ={this.bookmarkHandle} btnColor = {this.state.BookMarkBtnColor}/>
              </div>
              <div className='avatar-text-group'>
                <div className='user-profile-picture'>
                  <RoundedImage imageSource={userAvatarImage || '/assets/img/placeholder-profile-picture.png'} alt='profile picture'/>
                  <VerticalMargin className='hide-for-medium' size={50} />
                </div>
                <HorizontalMargin className='show-for-medium' size={10} />
                <div className='article-username-rate-group'>
                  <div className='article-header-username'>{username}</div>
                  <div className ='article-rating-container'>
                    <div className='article-header-rating'>
                      <RatingStars rateNumber={(averageRating)} />
                    </div>
                    <div className='article-header-rated-times'>
                      {!ratingsCount ? 'No ratings yet' : `rated by ${ratingsCount} user${ratingsCount > 1 ? 's' : ''}`}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className='body-section'>
          <div className='auto-container wide-padding'>
          <div className = "auto-container-body">
          <div className='impressions-container'>
              <div className='impression'>
                <CommentButtonGroup commentsCount = { commentsCount }/>
              </div>
              <div className='impression'>
                <LikeNumberGroup  onClick={this.likeHandle} btnColor = {userLiked && this.state.likeBtnColor} likeCount = {totalLikes || 0}/>
              </div>
              <div className='impression'>
                <DisLikeNumberGroup  onClick ={this.dislikeHandle} btnColor = {userDisliked && this.state.likeBtnColor} dislikeCount = {totalDislikes || 0}/>
              </div>
            </div>
            
            <div className='article-body'>
              <div dangerouslySetInnerHTML={{ __html: body }} />
            </div>
            </div>
            <div className="body-section-info-section">
            <HorizontalLine />
            <div className='section-article-info'>  
              <div>Rate this Article <RatingStarsBox starClickHandle= {this.starClickHandle} ratingNumber = {userRating || this.state.currentUserRating} /></div> 
              <div className='impressions-container-mobile'>
                <div className='impression'>
                  <CommentButtonGroup commentsCount = { commentsCount }/>
                </div>
                <div className='impression'>
                  <LikeNumberGroup  onClick={this.likeHandle} btnColor = {userLiked && this.state.likeBtnColor} likeCount = {totalLikes || 0}/>
                </div>
                <div className='impression'>
                <DisLikeNumberGroup  onClick ={this.dislikeHandle} btnColor = {userDisliked && this.state.likeBtnColor} dislikeCount = {totalDislikes || 0}/>
                </div>
              </div> 
              <VerticalMargin className='show-for-medium' size={10} />
              <div>
                <div className='section-comment-create'>
                  <div>Leave comments</div>
                  <CreateComment
                    articleId={articleId}
                    addComment={addComment}
                  />
                </div>
                <Spinner loading={loading} />
                <div className='section-user-comments'>
                  <CommentEntries
                  handleOnClickLike={(id) => this.handleOnClickLike(id)}
                  allCommentsImpressions={comments.commentsLikes}
                  activateModal={this.activateModal}
                  comments={ comments.comments.length ? comments.comments : articleComments } 
                  />
                </div>
              </div>
            </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SingleArticleView.propTypes = {
  articles: PropTypes.array.isRequired,
  comments: PropTypes.object,
  history: PropTypes.object,
  match: PropTypes.object.isRequired,
  fetchArticle: PropTypes.func.isRequired,
  rateArticle: PropTypes.func.isRequired,
  getArticleAvgRating: PropTypes.func.isRequired,
  removeArticleBookmark: PropTypes.func,
  addComment: PropTypes.func,
  getArticleComments: PropTypes.func,
  likeArticle: PropTypes.func,
  dislikeArticle: PropTypes.func,
  bookmarkArticle: PropTypes.func,
  getArticleLikes: PropTypes.func,
  getArticleDislikes: PropTypes.func,
  id: PropTypes.string,
  currentUserId: PropTypes.string,
  userLiked: PropTypes.any,
  userDisliked: PropTypes.any,
  impressions: PropTypes.object,
  userRating: PropTypes.number,
  averageRating: PropTypes.number,
  ratingsCount: PropTypes.number,
  getUserBookmarks: PropTypes.func,
  toggleModalOff: PropTypes.func,
  getAllCommentsImpression: PropTypes.func,
  likeComment: PropTypes.func,
  commentId: PropTypes.string,
  quickAuthAction: PropTypes.object,
  activateQuickAuthAction: PropTypes.func
}

const mapStateToProps = ({
  articles,
  comments,
  impressions = {},
  profile: { id: currentUserId },
  elementStatuses: { quickAuthAction },
}) => {
  const userLiked = !!impressions.likes.find(like => like.userId == currentUserId)
  const userDisliked = !!impressions.dislikes.find(disLike => disLike.userId == currentUserId)

  let foundRating = impressions.ratings.find(rating => rating.userId == currentUserId)
  const userRating = foundRating ? foundRating.rating : 0;

  const averageRating = impressions.ratings
    && impressions.ratings.reduce((totalRating, { rating }, i, { length }) => totalRating + rating/length, 0);
  
  const ratingsCount = impressions.ratings.length;

  return ({ 
    articles,
    comments,
    userLiked,
    userDisliked,
    userRating,
    currentUserId,
    impressions,
    averageRating,
    ratingsCount,
    quickAuthAction
  })
};

const mapDispatchToProps = ({
  fetchArticle: getOneArticle,
  rateArticle,
  getArticleAvgRating,
  addComment,
  getArticleComments,
  likeArticle,
  dislikeArticle,
  bookmarkArticle,
  getArticleLikes,
  getArticleDislikes,
  removeArticleBookmark,
  getUserBookmarks,
  getAllCommentsImpression,
  likeComment,
  activateQuickAuthAction
});

export default connect(mapStateToProps, mapDispatchToProps)(SingleArticleView);