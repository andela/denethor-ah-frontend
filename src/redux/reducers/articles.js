import { ADD_ARTICLE, REMOVE_ARTICLE } from '../actions/types'

const articleReducerDefaultState = [];

export default (state = articleReducerDefaultState, action) => {
  switch (action.type) {
    case ADD_ARTICLE:
      return [...state, action.article];

    case REMOVE_ARTICLE:
      return state.filter(({ id }) => id !== action.id );

    default:
      return state;
  }
};
