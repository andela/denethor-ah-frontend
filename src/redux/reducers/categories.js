import { GET_CATEGORIES_SUCCESS } from '../actions/types';

const categoryReducerDefaultState = [
	{
		id: 1,
		name: 'tech'
	},
	{
		id: 2,
		name: 'fashion'
	},
	{
		id: 3,
		name: 'lifestyle'
	},
	{
		id: 4,
		name: 'health'
	},
	{
		id: 5,
		name: 'science'
	},
	{
		id: 6,
		name: 'politics'
	},
];    

export default (state = categoryReducerDefaultState, action) => {
  switch (action.type) {
    
    case GET_CATEGORIES_SUCCESS:
        return [...action.payload];

    default:
      return state;
  }
};