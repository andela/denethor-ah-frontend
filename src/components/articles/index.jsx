import ArticlePage from './ArticlePage';
import ArticleCreatePage from './articleCreate/ArticleCreate';

export { ArticlePage, ArticleCreatePage };