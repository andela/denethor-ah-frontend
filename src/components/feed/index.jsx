import React from 'react';
import FeedBottomLargeCard from './FeedBottomLargeCard';
import FeedBottomSmallCard from './FeedBottomSmallCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getArticles } from '../../redux/actions/articles';
import { toast } from 'react-toastify';

class  FeedBottom extends React.Component {

  state = {
    selectedCategoryId: 0
  }

  filterArticle = (categoryName) => {

    const { categories } = this.props;
    const selectedCategory = categories.find(category => category.name === categoryName) || {};

    this.setState({
      selectedCategoryId: selectedCategory.id || 0
    });
  }

  componentDidMount() {
    const { categories } = this.props;
    categories.forEach(category => {
      this.props.getArticles(category.name)
        .then()
        .catch(function errorHandler() { toast.error('An error occurred while trying to get articles') })
    })
  }

  render() {
    const { articles, categories } = this.props;
    const _this = this;
    const { selectedCategoryId } = this.state;

    const filteredArticles = selectedCategoryId !== 0 ? articles.filter(article => article.categoryId === selectedCategoryId) : articles;

    const categoryFilterButtons = categories.slice(0,4).map(category => {
      return (
        <li key={category.id}><button className="category-buttons" onClick={function filter() { _this.filterArticle(category.name)} }>{category.name}</button></li>
      )
    });
    return (
      <div className="feed-bottom-section">
        <div className="nav-container-feed-bottom">
          <div className="top-categories-header"><h3>TOP CATEGORIES</h3></div>
          <div className="nav-items">
          <ul>
            <li><button className="category-buttons" onClick={function filter() { _this.filterArticle()} }>ALL</button></li>
            {categoryFilterButtons}
          </ul>
        </div>
        </div>

        <div className="feed-bottom">
        {
            filteredArticles && filteredArticles.slice(0,1).map((article, index) => (
              <FeedBottomLargeCard key={index} {...article} />
            ))
          }
        <div className="feed-bottom-side">
          <div className="feed-container-article-block-first">
          {
            filteredArticles && filteredArticles.slice(1,5).map((article, index) => (
              <FeedBottomSmallCard key={index} {...article} />
            ))
          }
          </div>
          <div className="feed-container-article-block-second">
          {
            filteredArticles && filteredArticles.slice(6,10).map((article, index) => (
              <FeedBottomSmallCard key={index} {...article} />
            ))
          }
          </div> 
        </div>
      </div>  
      </div>  
    )
  }
}

FeedBottom.propTypes = {
  getArticles: PropTypes.func.isRequired,
  articles: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired,
}

const mapStateToProps = state => ({
  articles: state.articles || [],
  categories: state.categories || []
})

export default connect(mapStateToProps, {getArticles})(FeedBottom);