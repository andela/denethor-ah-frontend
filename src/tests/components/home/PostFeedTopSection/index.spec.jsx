import React from 'react';
import { shallow } from 'enzyme';
import articles from '../../../mock-data/articles';
import categories from '../../../mock-data/categories';
import { PostFeedTopSection } from '../../../../components/feed/post-feed-top-section';

test('Should render post-feed-top-section with stats correctly', () => {
  const props = {
    articles,
    categories
  };

  const wrapper = shallow(
    <PostFeedTopSection {...props} />
  );
  
  expect(wrapper).toMatchSnapshot();
});