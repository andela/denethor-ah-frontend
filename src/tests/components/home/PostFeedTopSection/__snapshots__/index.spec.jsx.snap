// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render post-feed-top-section with stats correctly 1`] = `
<div
  className="auto-container"
>
  <div
    className="section-title"
  >
    Latest Publications
  </div>
  <div
    className="post-stream-top"
  >
    <PostCard
      article={
        Object {
          "author": Object {
            "imageUrl": "http://res.cloudinary.com/jsamcloud12/image/upload/v1552059783/profiles/2019-03-08T15:43:03.282Z.jpg",
            "username": "samcotech",
          },
          "body": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. <img src='site.com/image.jpg' /> Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
          "categoryId": 1,
          "comments": Array [
            Object {
              "articleId": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
              "commentBody": "Article comme",
              "createdAt": "2019-03-11T15:51:56.685Z",
              "id": "59df4232-0603-44a1-a560-7beffe5569fa",
              "userId": "b590e871-3df2-4cd2-b97f-569c3490b0fa",
            },
          ],
          "createdAt": "2019-03-07T21:05:17.071Z",
          "description": "this is the description",
          "id": "8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a",
          "likesCount": 20,
          "readTime": "1 minute",
          "references": null,
          "slug": "this-is-the-title",
          "title": "this is the title",
          "updatedAt": "2019-03-07T21:05:17.071Z",
        }
      }
      categories={
        Array [
          Object {
            "id": 1,
            "name": "tech",
          },
          Object {
            "id": 2,
            "name": "fashion",
          },
          Object {
            "id": 3,
            "name": "lifestyle",
          },
          Object {
            "id": 4,
            "name": "health",
          },
          Object {
            "id": 5,
            "name": "science",
          },
          Object {
            "id": 6,
            "name": "politics",
          },
        ]
      }
      key="8ee5e8ed-ecdf-41c1-9b94-6c6bb712a77a"
      size="big"
    />
  </div>
</div>
`;
